ShellPkg/Application/Shell/ShellProtocol.c

```diff
@@ -1817,16 +1830,32 @@ EfiShellExecute (
     return (EFI_UNSUPPORTED);
   }
 
+  Temp = NULL;
   if (NestingEnabled ()) {
     DevPath = AppendDevicePath (ShellInfoObject.ImageDevPath, ShellInfoObject.FileDevPath);
+    if (DevPath == NULL) {
+      return EFI_OUT_OF_RESOURCES;
+    }
 
     DEBUG_CODE_BEGIN ();
     Temp = ConvertDevicePathToText (ShellInfoObject.FileDevPath, TRUE, TRUE);
-    FreePool (Temp);
+    if (Temp != NULL) {
+      FreePool (Temp);
+    }
+
     Temp = ConvertDevicePathToText (ShellInfoObject.ImageDevPath, TRUE, TRUE);
-    FreePool (Temp);
-    Temp = ConvertDevicePathToText (DevPath, TRUE, TRUE);
-    FreePool (Temp);
+    if (Temp != NULL) {

```

Comment:
I'm not sure I understand what the original code is doing.
Device paths are converted to a text representation in `Temp`, then nothing is done with it ?

Reply:
Yeah, this is pretty weird. Pretty old, too: https://github.com/tianocore/edk2/commit/490ce43d92132653a6664971218e1cb330556391#diff-d3f247cd58398643eddd17d2024992818625e5251677fcfb11054df7b6200be1R1640. My best guess is that this was in there for testing that change (or maybe at point there was an assert or something in the `ConvertDevicePathToText()` path so this was verifying that this new environment could be created correctly.

I'd be fine to remove all the spurious code in the `DEBUG_CODE` block. Let me know what you think.

Reply:
I assume it can be let as is, if it's in a debug block ...